src/ds/address.h

```diff
@@ -19,16 +19,16 @@ namespace snmalloc
    * Perform pointer arithmetic and return the adjusted pointer.
    */
   template<typename T>
```

Comment:
How about the following:
```C++
  template <typename T = void>
  inline T* pointer_offset(void* base, size_t diff)
  {
    return reinterpret_cast<T*>(reinterpret_cast<char*>(base) + diff);
  }
```
I think this might be slightly cleaner,

Reply:
With `AuthPtr`, we're going to lose the ability to implicitly convert to `void*`.  How would you feel about

```c++
  template<typename U = void, typename T>
  inline U* pointer_offset(T* base, size_t diff)
  {
    return reinterpret_cast<U*>(reinterpret_cast<char*>(base) + diff);
  }
```

Reply:
I forget that you can put a default not on the last one in C++.  Yes, that is good.